src/hoc/with-analytic-on-event.tsx

```diff
@@ -1,23 +1,24 @@
 import * as React from 'react';
+import {Component, ComponentClass, ReactType, SyntheticEvent} from 'react';
 import * as PropTypes from 'prop-types';
 import {wrapDisplayName} from '../wrapDisplayName';
+import {ShisellContext} from '../shisell-context';
 
-import analyticsContextTypes, {AnalyticsContext} from '../analytics-context-types';
-
-export interface ExtraAnalyticsData {
-    [key: string]: any;
-}
 export interface ExtraAnalyticsDataProvider<Event> {
-    (event: Event): ExtraAnalyticsData;
+    (event: Event): {};
 }
-export type DataMapper<Event> = ExtraAnalyticsData | ExtraAnalyticsDataProvider<Event>;
-export type Predicate<T> = (val: T) => boolean;
+
+type DataMapper<Event> = {} | ExtraAnalyticsDataProvider<Event>;
+
+type Predicate<T> = (val: T) => boolean;
+
 export interface WithAnalyticOnEventConfiguration<TProps, TEvent> {
-    eventName: string;
+    eventName: keyof TProps;
```

Comment:
Passing in an event handler is optional - I think this will give a TS error if there's no prop called `{eventName}`, wouldn't it?